core/xml/src/main/java/org/opennms/core/xml/DoubleAdapter.java

```diff
@@ -42,9 +42,6 @@ public String marshal(final Double v) throws Exception {
         if (v == null) {
             return null;
         }
-        if (!v.isNaN() && !v.isInfinite() && v == Math.rint(v)) {
```

Comment:
Not sure if there is any reason to find and return integer values.  So fixed this as well

Reply:
I think there was a reason for doing this - iirc `rrdtool` will complain if we use any decimal point i.e. `1.0f` for counters.

Reply:
Looks like this is getting used only in [DiscoveryConfiguration](https://github.com/OpenNMS/opennms/blob/develop/opennms-config-model/src/main/java/org/opennms/netmgt/config/discovery/DiscoveryConfiguration.java#L67).  @RangerRick  also thinks it is safe to change this as it was done to mimic castor behavior as we no longer use castor anywhere.

Reply:
@j-white  Compared resource graphs on a node on Minion and a node on OpenNMS for counters specifically tcp* and ifHc* and they look similar.   So think we don't need to discard fractions like `x.0f` in marshalling.  Also unmarshalling converts numeric `x` -> `x.o` always.

Reply:
With rrdtool as well, resource graphs looks similar.